CMakeLists.txt

```diff
@@ -131,6 +131,26 @@ if(WIN32)
     # Improve parallelism in MSBuild.
     # See: https://devblogs.microsoft.com/cppblog/improved-parallelism-in-msbuild/.
     list(APPEND CMAKE_VS_GLOBALS "UseMultiToolTask=true")
+
+    # Production quality warning level.
+    try_append_cxx_flags("/W3" TARGET core)
+    # Disable warning C4018 "'token' : signed/unsigned mismatch".
```

Comment:
Do we really need to copy-paste the definitions of all these MSVC warning flags in here?

Reply:
At least, it does not hurt. It's also a nice way (at least for me) to document the silenced warnings, providing a quick understanding of their severity. In this PR, I followed the same practice that was accepted in the secp256k1 repo.

Anyway, I'm open to alternative approaches.

Reply:
It introduces a lot of verbosity, for something that anyone can easily lookup? I don't think we should do this specially for MSVC, if we aren't doing it for all other warning / compile / link flags?

Reply:
Thanks! Dropped.